cloud-controller-manager/do/loadbalancers.go

```diff
@@ -733,13 +737,24 @@ func buildHTTP3ForwardingRule(ctx context.Context, service *v1.Service, godoClie
 		return nil, errors.New("certificate ID is required for HTTP3")
 	}
 
+	var targetProtocol string
+	if targetProtocol, err = getTargetProtocol(service); err != nil {
+		return nil, err
+	}
+	if targetProtocol == protocolTCP || targetProtocol == protocolUDP {
```

Comment:
Do we need this explicit check still given that `getTargetProtocol()` now returns an error if a protocol other than from the allowed list (which does not include TCP or UDP) is specified?

Similar question for other occurrences of this check.